4-float-features/4-float-features.md

```diff
@@ -0,0 +1,74 @@
+# Introduction
+
+Num type has been separated into Int type and Float type.\
+The old Num type operators and features only support Int types, so there is a need to support Float types also.
+
+We propose the addition of basic operators to Float types to match those of Int types.\
+Also, we propose more support in recognizing Int and Float types, such as operator type prediction.\
+When adding these features, consistency will be the key.
+
+# Float Operators
+*overloading will be avoided until a more general consensus for overloading is set.
+
+## Boolean Float Operators
+
+There should be boolean operators for float types too.\
+Currently, there are 3 boolean operators, only supporting Int types.
+* `<`
+* `>`
+* `=`
+
+We propose the addition of the same operators, but for Float types.\
+In order to be consistent with Float Ops for plus, minus, and times, we will **Not** overload the operators*.\
+Instead, the operators will have a following **"."** after them.
+* `<.`
+* `>.`
+* `=.`
+
+## division operator
+
+There should be a division operator to complete the set of basic arithmetic operators.\
+There should be one operator `/` for Integer division and another for float division `/.` to avoid overloading*.\
+Also, casting will be avoided for now.
```

Comment:
We do need explicit conversion from int to float and vice versa though. These should be built-in functions. Check in with @annie-anna and @seanyeon since they are also adding built-in functions.

Reply:
The type of casting that I'm most familiar with have the syntax of (type) expression or type(expression). Is there a more "modern" way of casting that is preferred?

Reply:
That won't work well with structure editing because we already use parens, and the inside of parens has to be an expression rather than a type. 

We should just follow OCaml and have it be a built in function of type "float : Int -> Float" and "int : Float -> Int" that behave identically to the OCaml builtins `float_of_int` and `int_of_float` respectively. The way to do that is to make these things keywords, like `true` and `false` currently are. Unlike true and false, these are not valid patterns. So they are like `assert`, which @seanyeon is implementing.